eras/shelley/impl/src/Cardano/Ledger/Shelley/LedgerState.hs

```diff
@@ -1153,45 +1156,87 @@ startStep slotsPerEpoch b@(BlocksMade b') es@(EpochState acnt ss ls pr _ nm) max
       Coin rPot = _feeSS ss <> deltaR1
       deltaT1 = floor $ unboundRational (getField @"_tau" pr) * fromIntegral rPot
       _R = Coin $ rPot - deltaT1
+
+      -- We now compute stake pool specific values that are needed for computing
+      -- member and leader rewards.
+      activestake = fold . unStake $ stake'
       totalStake = circulation es maxSupply
+      mkPoolRewardInfoCurry =
+        mkPoolRewardInfo
+          (getField @"_d" pr, getField @"_a0" pr, getField @"_nOpt" pr)
+          _R
+          b
+          (fromIntegral blocksMade)
+          stake'
+          delegs'
+          totalStake
+          activestake
+
+      -- We map over the registered stake pools to compute the revelant
+      -- stake pool specific values.
+      allPoolInfo = Map.map mkPoolRewardInfoCurry poolParams
+
+      -- Stake pools that do not produce any blocks get no rewards,
+      -- but some information is still needed from non-block-producing
+      -- pools for the ranking algorithm used by the wallets.
+      blockProducingPoolInfo = Map.mapMaybe rightToMaybe allPoolInfo
+      rightToMaybe (Left _) = Nothing
+      rightToMaybe (Right x) = Just x
```

Comment:
You can import it from `Cardano.Prelude.rightToMaybe`, since `cardano-ledger-shelley` already depends on `cardano-prelude`, which reexports:

https://hackage.haskell.org/package/protolude-0.3.0/docs/Protolude.html#v:rightToMaybe

Reply:
fantastic, I was trying to avoid depending on etiher-combinators.